### PR TITLE
chore: bump angular to v19

### PR DESCRIPTION
**Pull request by Laffery**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Application (the showcase website) / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

- Bump deps version
	- Bump angular related packages and `icons-angular` to v19
	- `ngx-quicklink` (used in doc site) have no supported v19 yet, remove it from devDeps temporarily
	- Bump typescript to 5.6.x
	- Bump `@typescript-eslint/*` to v8 (cuz ts>5.6 is not supported by v7) and update some eslint rules 
	- Bump test related packages to latest version, such as jasmine, puppeteer
- Remove `standalone: true` beacause it become default value in v19
- Handle with some break changes in @angular/cdk, such as `CdkOverlayOrigin` no longer supports to pass argument to its constructor
- Add missing `imports` in the rest of test components, which never declared `standalone` property but was changed into standalone mode

## Does this PR introduce a breaking change?
- [x] Yes
- [ ] No

### CODE REVIEW

**Comment by HyperLife1119** [comment]
As far as I know, ngx-quicklink already [supports v19](https://github.com/mgechev/ngx-quicklink/commit/82eeebc1471e5a849d75a0dbed332301a954ba6d), but I still recommend to remove it completely to reduce dependencies, because I don't think it will be much of a benefit for a documentation site, the bundle size of the documentation pages is not that big, and no additional preloading strategy is needed.

**Comment by HyperLife1119** [comment]
Since this is a "huge" PR, I recommend leaving it until the end :)

**Comment by Laffery** [comment]
> As far as I know, ngx-quicklink already [supports v19](https://github.com/mgechev/ngx-quicklink/commit/82eeebc1471e5a849d75a0dbed332301a954ba6d), but I would still recommend removing it altogether and thus reducing the number of dependencies, as I don't think it's very beneficial to the documentation site, and the bundle size of the documentation pages isn't large enough to require an additional preloading strategy. WDYT?

Agree with you. The repo of `ngx-quicklink` is not quite activated, the package also brings us some License Compiliant issues. We can just remove it because its few benifit

**Comment by Laffery** [comment]
> As far as I know, ngx-quicklink already [supports v19](https://github.com/mgechev/ngx-quicklink/commit/82eeebc1471e5a849d75a0dbed332301a954ba6d), but I would still recommend removing it altogether and thus reducing the number of dependencies, as I don't think it's very beneficial to the documentation site, and the bundle size of the documentation pages isn't large enough to require an additional preloading strategy. WDYT?

Agree with you. The repo of `ngx-quicklink` is not quite activated, the package also brings us some License Compiliant issues. We can just remove it because its few benifit

File: .eslintrc.js [hunk 1]
```diff
@@ -67,28 +67,8 @@ module.exports = {
...
-                message: 'Use boolean instead.'
-              },
-              Function: {
-                message: 'Use specific callable interface instead.'
-              }
-            }
-          }
-        ],
+        '@typescript-eslint/no-unsafe-function-type': 'error',
+        '@typescript-eslint/no-wrapper-object-types': 'error',
```
Comment (Laffery) [inline .eslintrc.js:71]: ref https://typescript-eslint.io/blog/announcing-typescript-eslint-v8#replacement-of-ban-types

File: scripts/schematics/demo2schematics.ts [hunk 2]
```diff
@@ -38,7 +38,7 @@ describe('<%= classify(name) %>Component', () => {
     TestBed.configureTestingModule({
       declarations: [ <%= classify(name) %>Component ]
     })
-    .compileComponents();
+    ;
```
Comment (HyperLife1119) [inline scripts/schematics/demo2schematics.ts:41]:
```ts
 });
```

**Review by HyperLife1119 — CHANGES_REQUESTED** [review]

File: docs/universal.zh-CN.md
[the remark this answers is not in the document]
Comment (Laffery) [reply docs/universal.zh-CN.md:1]: 同意，之前我考虑重新写一篇基于 `@angular/ssr` 的文档，但其实在服务端渲染中 ng-zorro 并没有特别需要配置的，所以可以移除

**Review by HyperLife1119 — APPROVED** [review]
LGTM

**Pull request merged**